content/docs/engine-dev/tools/tiaf/maintaining-tiaf/debugging-tiaf.md

```diff
@@ -77,47 +87,179 @@ In the log above, you can extract the following information about the AR run and
+            "console": "integratedTerminal",
+            "justMyCode": true,
+            "args": [
+                "--config", "C:/dev/o3de/build/windows_vs2019/bin/TestImpactFramework/debug/Persistent/tiaf.json", // Replace with your build path to tiaf.json
+                "--src-branch", "development", // src and dst must be same in order to write out historic data
+                "--dst-branch", "development", // Note: the src and dst branches can be any name and do not need to refer to real, existing branches
+                "--commit", "4b65521188a2a0beb399eca7b692216d2fe3d209",
+                "--mars-index-prefix", "o3de-tiaf",
+                "--build-number", "001", // The build number can be anything
+                "--suites", "smoke", "main", // The suites must match those used in AR
+                "--label-excludes", "REQUIRES_gpu", // The label exclude filters must match those used in AR
+                "--test-failure-policy", "continue",
+                "--target-output", "stdout",
+                "--runtime-type", "native" // change to "python" for Python TIAF
+            ]
+        }
+    ]
+}
+```
+
+When the script has completed execution you will find the same historic data used in AR at the location `<build_path>/bin/TestImpactFramework/debug/Persistent/Native/historic/main-smoke`.
+
+{{< note >}}
+It is advised that you archive the contents of the `Persistent` directory as subsequent runs of CMake will delete the contents of this directory.
+{{< /note >}}
+
+### Step 2: Build and run the commit failing in AR
+
+Checkout the commit that is failing in AR (in this instance `85a0bd96daa4bd330d56bd817e8cdaf29d172b63`) and then run the TIAF using the AR scripts to generate the change list and arguments used by the runtime. It is important to set the `dst-branch` to the same branch used in the previous step but the `src-branch` to something different. This is for two reasons: firstly, it will ensure that we do not overwrite the historic data (as PR builds are not sources of truth and thus do not store their run data in the persistent storage) and secondly, it is required for generating change lists for unrelated previous and current commits (i.e. commits from unrelated branches).
+
+
+The script configuration file used for this step is as follows:
+
+```json
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Python: Current File",
+            "type": "python",
+            "request": "launch",
+            "program": "${file}",
+            "console": "integratedTerminal",
+            "justMyCode": true,
+            "args": [
+                "--config", "C:/dev/o3de/build/windows_vs2019/bin/TestImpactFramework/debug/Persistent/tiaf.json", // Replace with your build path
```

Comment:
Would it be safer and more siloed to use a different CMake build folder for gathering the historic data vs building the user commits that fails?

That way the user doesn't have to worry about archiving the contents of the `Persistent` directory if they use a different folder.

Since the the path to the `tiaf.json` can be supplied via the `--config` option, it can just be supplied from the data in the other build folder.

It also avoids any cross contamination of build artifacts that could cause per user environment issues.

Reply:
To provide an example.
Gathering the historic run data can be done in a cmake configure directory of `C:/dev/o3de/build/windows_tiaf_historic`, while the build to reproduce the test failure can be configured in a directory of `C:/dev/o3de/build/windows`.

That way there is a clear separation between the two and it is easy to cleanup one without affecting the other.
If the user needs to do a clean build of their branch with the test failure in it, they can just delete their `C:/dev/o3de/build/windows` directory and recreate it while still maintaining the persistent data

Reply:
One option would be to have a CMake var called something like `O3DE_TEST_IMPACT_CLEAR_RUNTIME_ARTIFACTS` that, when set, will clear the contents of the `Temp` and `Persistent` folders for each runtime. We could have this `TRUE` by default, and have the `profile` configs in `build_config.json` set this var to `TRUE`. Thoughts?

Reply:
>Would it be safer and more siloed to use a different CMake build folder for gathering the historic data vs building the user commits that fails?

The only issue I see with this is that the `tiaf_driver.py` script is designed such that each successive run of the script will update (read: replace) the existing historical data with the latest test run and coverage results. I'm not sure if there is a practical way to have the separation between "last" and "current" without having to ping-pong between the two folders for each run (clearing the "current" prior to invoking the runtime) which seems like it would be contrary to the philosophy of keeping those separate.

Reply:
We discussed a command lien arg to be added, I will update this PR with a suggestion for how to preserve the historic data locally using the existing config files, @lumberyard-employee-dm is going to add a new command line arg to override the persistent storage location as a quality of life improvement and open up a PR with the updated docs when that improvement is in 👍